Source/ACE.Server/WorldObjects/Portal.cs

```diff
@@ -134,6 +134,77 @@ public override ActivationResult CheckUseRequirements(WorldObject activator)
                     // You are too powerful to interact with that portal!
                     return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.YouAreTooPowerfulToUsePortal));
                 }
+
+                var playerPkLevel = player.PkLevel;
+
+                if (PropertyManager.GetBool("pk_server").Item)
+                    playerPkLevel = PKLevel.PK;
+                else if (PropertyManager.GetBool("pkl_server").Item)
+                    playerPkLevel = PKLevel.PKLite;
+
+                if ((PortalRestrictions & PortalBitmask.Undef) != 0)
+                {
+                    // Players may not interact with that portal.
+                    return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.PlayersMayNotUsePortal));
+                }
+
+                if ((PortalRestrictions & PortalBitmask.NoPk) != 0 && (playerPkLevel == PKLevel.PK || player.PlayerKillerStatus == PlayerKillerStatus.PK))
+                {
+                    // Player killers may not interact with that portal!
+                    return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.PKsMayNotUsePortal));
+                }
+
+                if ((PortalRestrictions & PortalBitmask.NoPKLite) != 0 && (playerPkLevel == PKLevel.PKLite || player.PlayerKillerStatus == PlayerKillerStatus.PKLite))
+                {
+                    // Lite Player Killers may not interact with that portal!
+                    return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.PKLiteMayNotUsePortal));
+                }
+
+                if ((PortalRestrictions & PortalBitmask.NoNPK) != 0 && playerPkLevel == PKLevel.NPK)
+                {
+                    // Non-player killers may not interact with that portal!
+                    return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.NonPKsMayNotUsePortal));
+                }
+
+                if ((PortalRestrictions & PortalBitmask.OnlyOlthoiPCs) != 0
+                    && player.WeenieClassId != 43480 && player.WeenieClassId != 43481
```

Comment:
These should either be using (int)WeenieClassName.ProperType, or there should be a comment about what exactly '43480' etc. is at the bare minimum